### PR TITLE
acme: registration improvements

### PR DESCRIPTION
**Pull request by kuba**
This is a prelude to the account overhaul PR.

r? @jdkasten

### CODE REVIEW

File: acme/messages.py [hunk 1]
```diff
@@ -182,15 +182,27 @@ def emails(self):
 
     @property
     def phone(self):
-        """Phone."""
-        assert len(self.phones) == 1
-        return self.phones[0]
+        """Phone.
```
Comment (jdkasten) [inline acme/messages.py]: This doesn't seem right.  I don't know when it would be useful to pick any phone or any email?

Comment (kuba) [inline acme/messages.py]: https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/client.py#L126-128

I guess, I will have to fix that one as well...

Comment (kuba) [inline acme/messages.py]: There are many other places that rely on just singular email and it doesn't make sense to fix it now, as my next PR just scrambles everything. I will make sure to remove singulars in next PR.

Comment (jdkasten) [inline acme/messages.py]: That is fine.

**Pull request merged**